apps/web/src/components/WalletConnect/SessionProposalModal.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  Box,
+  Card,
+  FormControl,
+  FormErrorMessage,
+  HStack,
+  ModalContent,
+  Text,
+  VStack,
+  useToast,
+} from "@chakra-ui/react";
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import { useAsyncActionHandler, useGetImplicitAccount, walletKit } from "@umami/state";
+import { buildApprovedNamespaces, getSdkError } from "@walletconnect/utils";
+import { useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+import RequestModal from "./RequestModal";
+import { CheckmarkIcon, CloseIcon } from "../../assets/icons";
+import { OwnedImplicitAccountsAutocomplete } from "../AddressAutocomplete";
+
+/**
+ * Finds the overlap between the chains supported by the wallet and the chains requested by the dApp.
+ * We take the first required chain as Umami supports connection for only one chain per request.
+ */
+function getChainToConfirm(required: (string | string[] | undefined)[]) {
+  // take the first required chain which is supported by the wallet
+  for (const chain of required.flat()) {
+    if (chain && ["tezos:mainnet", "tezos:ghostnet"].includes(chain)) {
+      return chain;
+    }
+  }
+  return undefined;
+}
+
+export default function SessionProposalModal({
+  proposal,
+}: {
+  proposal: WalletKitTypes.SessionProposal;
+}) {
+  console.log("SessionProposalModal", proposal);
+  const getAccount = useGetImplicitAccount();
+  const toast = useToast();
+
+  const { onClose } = useDynamicModalContext();
+  const { handleAsyncAction } = useAsyncActionHandler();
+
+  const [isLoadingApprove, setIsLoadingApprove] = useState(false);
+  const [isLoadingReject, setIsLoadingReject] = useState(false);
+  const form = useForm<{ address: string }>({
+    mode: "onBlur",
+  });
+  const {
+    getValues,
+    formState: { errors, isValid },
+  } = form;
+
+  const required = Object.entries(proposal.params.requiredNamespaces).map(([key, values]) =>
+    key.includes(":") ? key : values.chains
+  );
+
+  if (required.length !== 1) {
+    console.error("Expected only one required chain", required);
```

Comment:
If we have this check for only having 1 required network, should we change `getChainToConfirm()` to accept one chain instead of the list?

Reply:
Product question: If there is a list of requested chains that matches more than one Umami network - should we allow user to choose which network they'd like to use for the connection?

Reply:
if the user requests more than one required network, then the answer confirming just one network will be rejected by the WalletConnect SDK. So either Umami starts supporting multiple networks in one request (complex task) or dApps must request exactly one network (ok for now?)

Reply:
Ah, I see 🤔 

@dianasavvatina Could you please add a comment in the code (that all required networks should be supported on umami side for the request to pass)?

IMO just one network should be OK (at least for now), but worth checking with the rest of the team